sklearn/preprocessing/label.py

```diff
@@ -33,6 +34,76 @@
 ]
 
 
+def _nanunique(ar):
+    uniques = np.unique(ar)
+    # np.nan is always sorted last
+    if is_scalar_nan(uniques[-1]):
+        nan_idx = np.searchsorted(uniques, np.nan)
+        return uniques[:nan_idx+1]
+    else:
+        return uniques
+
+
+def _nanencode_numpy(values, uniques=None, encode=False,
+                     missing_value=np.nan):
```

Comment:
for good or bad, this should be `missing_values` not `missing_value` for consistency.